tests/unit/s2n_mem_usage_test.c

```diff
@@ -33,9 +33,18 @@
  * usage. The greater the value, the more accurate the end result. */
 #define MAX_CONNECTIONS 1000
 
+#ifdef OPENSSL_IS_AWSLC
+/* s2n + AWS-LC memory usage is currently a little higher than s2n + OpenSSL due to:
+    * AWS-LC uses EVP_CIPHER_CTX contexts as well as EVP_AEAD_CTX in s2n_session_key (this could be a union in the future)
+    * AWS-LC uses an HMAC_CTX which is larger than OpenSSL EVP_PKEY in s2n_evp_hmac_state without being able to get rid
+        * of the EVP_MD_CTX in s2n_evp_digest because s2n_hash uses that type
+    * AWS-LC FIPS uses Jitter for entropy which requires additional state per thread
+*/
+#define MEM_PER_CONNECTION (49 * 1024)
+#else
```

Comment:
Can you avoid branching here? FIPS mode is usually higher. We just set MEM_PER_CONNECTION to the higher value. This is intended as a ceiling / sanity check, not as an exact number.

We should be really careful with this work though. The last round of FIPs work was what caused the connection memory to balloon in the first place.

Reply:
I agree, I created https://github.com/aws/s2n-tls/issues/3114 for follow up and got rid of the branch.

Reply:
Turns out the union changes were easy to add, and cleaning up the functions that use the hmac_impl the connection size actually got a bit smaller!